impeller/renderer/backend/vulkan/pipeline_cache_data_vk.cc

```diff
@@ -0,0 +1,132 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "impeller/renderer/backend/vulkan/pipeline_cache_data_vk.h"
+
+#include "flutter/fml/file.h"
+#include "impeller/base/allocation.h"
+#include "impeller/base/validation.h"
+
+namespace impeller {
+
+static constexpr const char* kPipelineCacheFileName =
+    "flutter.impeller.vkcache";
+
+// Inspired by
+// https://medium.com/@zeuxcg/creating-a-robust-pipeline-cache-with-vulkan-961d09416cda.
+struct PipelineCacheHeader {
+  // This can be used by Impeller to manually invalidate all old caches.
+  uint32_t magic = 0xC0DEF00D;
```

Comment:
@jtmcdole this is almost you

Reply:
lol yeah! The other one I had in mind was `0xFA51F00D` (fast-food for our pipelines) but the `T` didn't read right.